src/ORM/DataObject.php

```diff
@@ -640,7 +640,8 @@ protected function duplicateManyManyRelation($sourceObject, $destinationObject,
             $extraFieldNames = [];
         }
 
-        foreach ($source as $item) {
+        // Disabling sort improves performance
+        foreach ($source->sort(null) as $item) {
```

Comment:
This loop just adds items to a `DataList`.
The order in which items are added doesn't matter.